packages/client/src/components/TrackList/TrackList.jsx

```diff
@@ -0,0 +1,268 @@
+import React, { useEffect, useState } from "react";
+import { DragDropContext, Droppable } from "react-beautiful-dnd";
+import { motion } from "framer-motion";
+import { sortArrayAscendent, sortArrayDescendent } from "../../utils/sorters";
+import { containerAnimation } from "../../utils/motionSettings";
+import TrackCard from "../TrackCard";
+import TrackSorter from "../TrackSorter/TrackSorter";
+
+function TrackList({ tracks, onAddLikedColumn }) {
+  const [listOfTracks, setListOfTracks] = useState(tracks);
+  const [flag, setFlag] = useState({
+    flagTitle: 1,
+    flagAlbum: 1,
+    flagPopularity: 1,
+    flagDuration: 1,
+  });
+
+  const handleAddLikedColumn = (song, isLiked) => {
+    onAddLikedColumn(song, isLiked);
+  };
+
+  const handleDeletedView = (trackId) => {
+    const updatedListOfTracks = listOfTracks.filter((v) => v._id !== trackId);
+    setListOfTracks(updatedListOfTracks);
+  };
+
+  const sortByTitleAndArtist = () => {
```

Comment:
In this function I wouldn't use numbers to specify different states, maybe a named state could be better, with an array of used states